File: Level1/행렬의 덧셈/main.py
# Link : https://programmers.co.kr/learn/courses/30/lessons/12950

## 나의 풀이 ##
def solution(arr1, arr2):
    answer = [] #answer[[]]로 생성하면 append 할 때 첫 번째 행으로 []가 생성 되어서 자꾸 에러가 뜬다! 그래서 []로 생성! 

    for i in range(len(arr1)): # 행 개수만큼 반복 
        line = [] # 빈 리스트 생성
        for j in range(len(arr1[0])): # 열 개수만큼 반복 
            line.append(arr1[i][j] + arr2[i][j]) #추가하고
        answer.append(line) # 한 행을 추가        

    return answer
  
  
  ## 다른 사람의 풀이 ##
  # Link : https://programmers.co.kr/learn/courses/30/lessons/12950/solution_groups?language=python3
  
def sumMatrix(A,B):
    answer = [[c + d for c, d in zip(a, b)] for a, b in zip(A,B)] 
    # zip 함수는 두 그룹의 데이터를 서로 엮어주는 파이썬의 내장함수! 
    # a, b는 리스트(한 행)를 의미하고 c, d는 리스트 안의 요소를 의미하는 듯!
    return answer
  
# 아래는 테스트로 출력해 보기 위한 코드입니다.
# 출력값은 [[4,6], [7,9]]
print(sumMatrix([[1,2], [2,3]], [[3,4],[5,6]]))

File: Level1/행렬의 덧셈/main.cpp
// Link : https://programmers.co.kr/learn/courses/30/lessons/12950

#include <string>
#include <vector>
#include <iostream>

using namespace std;

vector<vector<int>> solution(vector<vector<int>> arr1, vector<vector<int>> arr2) {
    vector<vector<int>> answer; // 답이 되는 2차원 벡터
    vector<int> answer1; // 1차원 벡터 즉 행의 한 벡터를 담당할 친구들이다

    for(int i= 0; i < arr1.size(); i++){ // 행의 개수 담당
        for(int j=0; j < arr1[0].size(); j++){ // 열의 개수 담당
            answer1.push_back(arr1[i][j] + arr2[i][j]); 
            // 인덱스로 접근할 수 있는 건 원소가 있을 때만. 메모리에 할당이 안 되었으므로 push_back을 사용
            // 한 행에 대한 벡터 배열을 저장하고 
        }
        answer.push_back(answer1); // 한 행을 전체 벡터의 한 원소로서 저장! 
        answer1.clear(); //초기화 push_back은 마지막 원소 다음에 저장하기 때문에 초기화 시켜주장
    }
    return answer;
}
